ibl_to_nwb/brainwide_map/read_after_write.py

```diff
@@ -0,0 +1,361 @@
+import os
+import numpy as np
+import pandas as pd
+from pathlib import Path
+import h5py
+from pynwb import NWBHDF5IO, NWBFile
+from one.api import ONE
+import logging
+from typing import Optional
+from numpy import testing
+
+def check_arrays(array_a: np.ndarray, array_b: np.ndarray, full_check: bool = False):
+    """checks if two arrays contain the same numerical values
+
+    Args:
+        array_a (np.ndarray): _description_
+        array_b (np.ndarray): _description_
+        full_check (bool, optional): If True, compares all values of the arrays. If False, checks a small subsample only. Defaults to False.
+    """
+
+    # if full check, check all samples
+    if full_check:
+        testing.assert_allclose(array_a, array_b)
+
+    # check just a random subset of samples
+    else:
+        inds = np.random.randint(0, np.prod(array_a.shape), size=10)
+        testing.assert_allclose(np.ravel(array_a)[inds], np.ravel(array_b)[inds])
+        
+
+
+def check_series(series_a: pd.Series, series_b: pd.Series, full_check: bool = False):
```

Comment:
Again, not really recommended to define a one-line redirection function. Just making more work than necessary for ourselves here

Reply:
This had somewhat historical reasons and I left it in as a stub. When I started coding, I wasn't even aware of the existence of `numpy.testing` so the initial flow of the data was in chained calls where I could go starting from a complex datatype like a `pd.DataFrame` consecutively to simple ones, where in the end an `assert()` call would take care of the comparison of the selected numbers.

Things have changed, but I left the general structure up for discussion in case it could still be useful for a case I haven't thought about.